hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/wal/AbstractFSWAL.java

```diff
@@ -907,6 +959,9 @@ public void shutdown() throws IOException {
     rollWriterLock.lock();
     try {
       doShutdown();
+      if (logArchiveExecutor != null) {
```

Comment:
Why not put this inside the doShutdown?

Reply:
doShutDown is abstract method. So its better we do in the Parent class.